Doc/library/compression.zstd.rst

```diff
@@ -0,0 +1,768 @@
+
+   The *samples* argument (an iterable of bytes), contains sample data for
+   generating the Zstandard dictionary.
+
+   The *dict_size* argument, an integer, is the maximum size (in bytes) the
+   Zstandard dictionary should be. Please see :func:`train_dict` for
+   suggestions on the maximum dictionary size.
+
+   The *level* argument (an integer) is the compression level expected to be
+   passed to the compressors using this dictionary. The dictionary information
+   varies for each compression level, so tuning for the proper compression
+   level can make compression more efficient.
+
+
+.. class:: ZstdDict(dict_content, /, *, is_raw=False)
+
+   A wrapper around Zstandard dictionaries. Dictionaries can be used to improve
+   the compression of many small chunks of data. Use :func:`train_dict` if you
+   need to train a new dictionary from sample data.
+
+   The *dict_content* argument (a :term:`bytes-like object`), is the already
+   trained dictionary information.
+
+   The *is_raw* argument, a boolean, is an advanced parameter controlling the
+   meaning of *dict_content*. ``True`` means *dict_content* is a "raw content"
+   dictionary, without any format restrictions. ``False`` means *dict_content*
+   is an ordinary Zstandard dictionary, created from Zstandard functions, e.g.
+   :func:`train_dict` or the ``zstd`` CLI.
+
+    .. attribute:: dict_content
+
+        The content of the Zstandard dictionary, a ``bytes`` object. It's the
+        same as the *dict_content* argument in the ``__init__`` method. It can
+        be used with other programs, such as the ``zstd`` CLI program.
+
+    .. attribute:: dict_id
+
+        Identifier of the Zstandard dictionary, a int value between 0 and .
+
+        Non-zero means the dictionary is ordinary, created by Zstandard
+        functions and following the Zstandard format.
+
+        ``0`` means a "raw content" dictionary, free of any format restriction,
+        used for advanced users.
+
+        .. note::
+
+            The meaning of ``0`` for :attr:`ZstdDict.dict_id` is different from
+            the ``dictionary_id`` argument to the :func:`get_frame_info`
+            function.
+
+    .. attribute:: as_digested_dict
+
+        Load as a digested dictionary, see below.
+
+    .. attribute:: as_undigested_dict
+
+        Load as an undigested dictionary.
+
+        Digesting a dictionary is a costly operation. These two attributes can
+        control how the dictionary is loaded to the compressor, by passing them
+        as the ``zstd_dict`` argument, e.g.
+        ``compress(data, zstd_dict=zd.as_digested_dict)``.
+
+        If don't use one of these attributes, an **undigested** dictionary is
+        passed by default.
+
+        .. list-table:: Difference for compression
+            :widths: 12 12 12
+            :header-rows: 1
+
+            * -
+              - | Digested
+                | dictionary
```

Comment:
Is the use of line-blocks (`|`) intentional?

Reply:
Not particularly no. This is based on the table in the pyzstd docs. I think these should probably be removed (I'll push a commit later to do so).